framework/src/media/FileInputDataSource.cpp

```diff
@@ -117,8 +117,8 @@ ssize_t FileInputDataSource::read(unsigned char *buf, size_t size)
 	std::shared_ptr<Decoder> decoder = getDecoder();
 
 	if (!buf) {
-		meddbg("buf is nullptr, hence return 0\n");
-		return 0;
+		meddbg("buf is nullptr, hence return ERROR\n");
```

Comment:
usually, caps ERROR indicates returning -1 . in my opinion, better to use small case to say just error.
but function is actually returning EOF

Reply:
Changed to EOF =)